pkg/ingester/ingester_test.go

```diff
@@ -9720,10 +9720,21 @@ func TestGetIgnoreSeriesLimitForMetricNamesMap(t *testing.T) {
 // The correctness of changed runtime is already tested in Prometheus, so we only check if the
 // change is being applied here.
 func Test_Ingester_OutOfOrder(t *testing.T) {
+	for name, tc := range ingesterSampleTypeScenarios {
```

Comment:
The one thing I'm missing from this ooo test (and this has already been a problem before this PR) is that they don't check the out of order related metrics of the ingester.

So I think we should add some checks on 
```
cortex_ingester_tsdb_out_of_order_samples_appended_total
cortex_ingester_tsdb_sample_out_of_order_delta_seconds
```

Reply:
Added checks for these metrics in https://github.com/grafana/mimir/pull/7175/commits/71927e7c76c5c99a206b5c9e285db10a093d4b8a